shell/server/feature-key.js

```diff
@@ -73,3 +74,252 @@ loadSignedFeatureKey = function (buf) {
+      // here, then we should take this chance to update to the current key.
+      if (!options.dryRun || parseInt(data.expires) > parseInt(key.expries)) {
+        const newTextBlock = fetchUpdatedFeautureKeyFromVendor(key);
+        setNewFeatureKeyInternal(db, newTextBlock, () => {
+          throw new Error("Renewal seemed to succeed but the service returned an expired key. " +
+                          "Is your system clock wrong?");
+        });
+      }
+    } else {
+      reportRenewalProblem(db, key, options, { unknownResponse: response.content });
+    }
+  } catch (err) {
+    console.error("Exception when trying to renew feature key:", err.stack);
+    reportRenewalProblem(db, key, options, { exception: err.message });
+  }
+};
+
+function fetchUpdatedFeautureKeyFromVendor(key) {
+  const fetchResponse = HTTP.get(RENEW_API_HOST + "/keys/" + key.secret.toString("hex"), {
+    headers: {
+      "Authorization": "Bearer " + RENEW_API_TOKEN,
+    },
+  });
+  const result = fetchResponse.data.key;
+  if (!result) throw new Error("Renewal server returned invalid GetKeyResponse.");
+  return result;
+}
+
+function reportRenewalProblem(db, key, options, problem) {
+  db.collections.featureKey.update("currentFeatureKey", { $set: { renewalProblem: problem } });
+
+  if (!options.interactive) {
+    if (key.isTrial) {
+      db.sendAdminNotification("trialFeatureKeyExpired", "/admin/feature-key");
+    } else {
+      db.sendAdminNotification("cantRenewFeatureKey", "/admin/feature-key");
+    }
+
+    const emailOptions = {
+      from: db.getReturnAddress(),
+      subject: key.isTrial
+          ? `URGENT: Sandstorm for Work trial for ${db.getServerTitle()} has expired`
+          : `URGENT: Couldn't renew Sandstorm for Work subscription for ${db.getServerTitle()}`,
+    };
+
+    emailOptions.text = key.isTrial
+        ? "This is an automated message from your Sandstorm server. Your trial of Sandstorm for Work has expired. To continue using Sandstorm for Work, update your subscription here:"
+        : "This is an automated message from your Sansdtorm server. There was an error when trying to renew your Sandstorm for Work subscription. To resolve the issue, please go to:";
+
+    emailOptions.text += `
+
+${process.env.ROOT_URL}/admin/feature-key`;
+
+    Meteor.users.find({ isAdmin: true }).forEach((user) => {
+      const email = _.findWhere(SandstormDb.getUserEmails(user), { primary: true });
+      if (!email) {
+        console.error("No email found for admin with userId:", user._id);
+        return;
+      }
+
+      try {
+        emailOptions.to = email.email;
+        send(emailOptions);
+      } catch (err) {
+        console.error(
+          `Failed to send deletion email to admin (id=${user._id}) with error: ${err}`);
+      }
+    });
+  }
+}
+
+let keyRenewalTimeout = null;
+let keyObserver = null;
+let currentlyRenewing = false;
+keepFeatureKeyRenewed = function (db) {
```

Comment:
Consider moving this whole file to `/imports` and importing this function in 00-startup.js